cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -312,17 +322,110 @@ public void testExternalSparkProgramPipelines() throws Exception {
   }
 
   @Test
-  public void testMacroActionPipelines() throws Exception {
+  public void testMacroActionPipelinesAndSchedule() throws Exception {
```

Comment:
don't hijack another unit test like this. Create your own that sets up two pipelines and has a logical flow. What you've done here is some weird mix that tests macros like before, but tests triggered stuff on one half of the pipelines with setup spread across multiple methods, making it difficult to tell what is being tested.